sample-app/src/utils/composeCssClasses.tsx

```diff
@@ -0,0 +1,60 @@
+import { composeTheme } from '@css-modules-theme/core';
+import { Compose, Theme } from '@css-modules-theme/core';
+
+/**
+ * The method of combining a component's built-in CSS classes with custom CSS classes
+ * @remarks
+ * Merge keeps the component's built-in classes and adds the custom classes to them (default).
+ * Replace ignore all of the component’s built-in classes and only uses the custom classes.
+ * Assign keeps the component's built-in classes, however custom classes will completely override their associated built-in classes.
+ */
+export type Composition = 'merge' | 'replace' | 'assign';
+
+/**
+ * Combines a component's built-in CSS classes with custom CSS classes.
+ * @param builtInClasses The component's built-in css classes
+ * @param customClasses The custom classes to combine with the built-in ones
+ * @param composition The method of combining the built-in classes with the custom classes
+ * @returns The composed CSS classes
+ */
+export function composeCssClasses<ClassInterface> (
```

Comment:
It's just a generic. I can switch to `T` since I think that would make it clearer

Reply:
Should the generic extend some interface or does it really let you pass anything?

Reply:
oh I'm also cool calling it `ClassInterface` instead of just `T` which is not as descriptive! My brain just derped out and didn't recognize it as a generic

Reply:
The `ClassInterface` should be an object with only strings as keys and strings as values (essentially a `Record<string,string>`). However using that Record causes an issue for Typescript because interfaces with a specific set of keys cannot be indexed by arbitrary strings. Each class interface will have an interface with a specific set of keys, and that's not compatible with `Record<string,string>`. I created the `isThemeObject` method so that I could confirm that the `builtInClasses` and the `customClasses` objects can be safely indexed by any string. The `Theme` object from css-modules-theme is simply `interface Theme { [key: string]: string; }`.